mantidimaging/gui/windows/spectrum_viewer/view.py

```diff
@@ -334,3 +384,40 @@ def set_binning_visibility(self) -> None:
         self.bin_size_spinBox.setHidden(hide_binning)
         self.bin_step_label.setHidden(hide_binning)
         self.bin_step_spinBox.setHidden(hide_binning)
+
+    def set_roi_properties(self) -> None:
+        current_roi = self.presenter.model.get_roi(self.current_roi)
+        self.roiPropertiesGroupBox.setTitle(self.roiPropertiesGroupBoxTitle + self.current_roi)
+        roi_iter_order = ["Left", "Top", "Right", "Bottom"]
+        row = 0
+        for pos in current_roi.__iter__():
+            with QSignalBlocker(self.roiPropertiesSpinBoxes[roi_iter_order[row]]):
+                self.roiPropertiesSpinBoxes[roi_iter_order[row]].setValue(pos)
+            row = row + 1
+        self.set_roi_spinbox_ranges()
+        self.presenter.redraw_spectrum(self.current_roi)
+        roi_width = abs(self.roiPropertiesSpinBoxes["Right"].value() - self.roiPropertiesSpinBoxes["Left"].value())
+        self.roiPropertiesLabels["Width"].setText(str(roi_width))
+        roi_height = abs(self.roiPropertiesSpinBoxes["Top"].value() - self.roiPropertiesSpinBoxes["Bottom"].value())
+        self.roiPropertiesLabels["Height"].setText(str(roi_height))
+        roi_area = roi_width * roi_height
+        self.roiPropertiesLabels["Area"].setText(str(roi_area))
+
+    def adjust_roi(self) -> None:
+        roi_iter_order = ["Left", "Top", "Right", "Bottom"]
+        new_points = [self.roiPropertiesSpinBoxes[prop].value() for prop in roi_iter_order]
+        new_roi = SensibleROI().from_list(new_points)
+        self.presenter.model.set_roi(self.current_roi, new_roi)
+        self.spectrum.adjust_roi(new_roi, self.current_roi)
+
+    def set_roi_spinbox_ranges(self):
+        spin_boxes = self.roiPropertiesSpinBoxes
+        for key, value in spin_boxes.items():
+            if key == "Left":
+                value.setMaximum(spin_boxes["Right"].value() - 1)
+            elif key == "Right":
+                value.setMinimum(spin_boxes["Left"].value() + 1)
+            elif key == "Top":
+                value.setMaximum(spin_boxes["Bottom"].value() - 1)
+            elif key == "Bottom":
+                value.setMinimum(spin_boxes["Top"].value() + 1)
```

Comment:
Maybe a silly question, but it this needed? I tried commenting out entirely including references and all seemed to perform fine.
If it is needed, you may be able to condense to something like:
```python
        self.roiPropertiesSpinBoxes["Left"].setMaximum(self.roiPropertiesSpinBoxes["Right"].value() - 1)
        self.roiPropertiesSpinBoxes["Right"].setMinimum(self.roiPropertiesSpinBoxes["Left"].value() + 1)
        self.roiPropertiesSpinBoxes["Top"].setMaximum(self.roiPropertiesSpinBoxes["Bottom"].value() - 1)
        self.roiPropertiesSpinBoxes["Bottom"].setMinimum(self.roiPropertiesSpinBoxes["Top"].value() + 1)
```

Reply:
This is mainly to prevent the user from selecting for example Top > Bottom via the spinboxes as it doesnt make much sense. Doing so doesn't break anything but it does give errors in the terminal. 
![image](https://github.com/mantidproject/mantidimaging/assets/30868085/bb5a1f10-0ec5-4220-aacc-3d3cfc5b9fe3)
![image](https://github.com/mantidproject/mantidimaging/assets/30868085/a4081042-1391-4b75-b8c3-4bb15f1328c0)

I think its worth keeping but I can condense it down.